### PR TITLE
feat(aggregations): add refresh document count in aggregation results COMPASS-6156

### PR DESCRIPTION
**Pull request by mabaasit**
Add a refresh button to re-count the number of documents in aggregation results.

https://user-images.githubusercontent.com/1305718/208673995-d03c6158-ff1a-49d4-8337-756084db70e5.mov

### Checklist
- [x] New tests and/or benchmarks are included
- [ ] Documentation is changed or added
- [ ] I have signed the MongoDB Contributor License Agreement (https://www.mongodb.com/legal/contributor-agreement)

## Motivation and Context

- [ ] Bugfix
- [x] New feature
- [ ] Dependency update
- [ ] Misc

## Types of changes

- [ ] *Backport Needed*
- [ ] Patch (non-breaking change which fixes an issue)
- [x] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: packages/compass-crud/src/components/crud-toolbar.tsx [hunk 1]
```diff
@@ -194,7 +194,8 @@ const CrudToolbar: React.FunctionComponent<CrudToolbarProps> = ({
         </div>
         <div className={toolbarRightActionStyles}>
           <Body data-testid="crud-document-count-display">
-            {start} - {end} of {displayedDocumentCount}
+            {start} – {end}{' '}
+            {displayedDocumentCount && `of ${displayedDocumentCount}`}
```
Comment (mabaasit) [inline packages/compass-crud/src/components/crud-toolbar.tsx:198]: drive-by to hide `of` when refreshing Documents Tab.

Comment (mabaasit) [reply packages/compass-crud/src/components/crud-toolbar.tsx:198]: and uses `–` instead of `-`. we are using this on aggregations screen and it looks better. and that's why some changes in e2e tests

File: packages/compass-aggregations/src/components/pipeline-results-workspace/pipeline-pagination-count.tsx [hunk 2]
```diff
@@ -29,32 +32,67 @@ const countButtonStyles = css({
...
-      aggregation again. This will be the equivalent of adding a
-      $count as the last stage of the pipeline.
-    `;
+const countDocsContainerStyles = css({
+  display: 'flex',
+  alignItems: 'center',
+});
+
+// As we want to align the refresh button and loader icon,
+// we need to set the same size as the IconButton.
```
Comment (addaleax) [inline packages/compass-aggregations/src/components/pipeline-results-workspace/pipeline-pagination-count.tsx:41]: This is referring to the IconButton LG type, whose styles I guess we don’t control?

**Review by addaleax — APPROVED** [review]

File: packages/compass-aggregations/src/components/pipeline-results-workspace/pipeline-pagination-count.tsx (hunk 2 above)
Comment (mabaasit) [reply packages/compass-aggregations/src/components/pipeline-results-workspace/pipeline-pagination-count.tsx:41]: Yes. And this is mainly to avoid flicker when showing Refresh Button and Spinner.
